docs/rfcs/0001-rfc-process.md

```diff
@@ -0,0 +1,63 @@
+- Status: Proposed
+- Date: 2023-05-23
+- RFC PR: TODO
+
+
+## Summary
+Introducing a process for proposing, reviewing, and documenting decisions with high-impact implications for our codebase.
+
+## Motivation
+
+RFCs authored by this team are currently submitted and reviewed in a number of places, namely Confluence and Google Docs.
+
+Several challenges have been noted with this practice:
+- Docs are accessible to Mozillians only
+- Platform constraints make asynchronous discussions difficult and impossible to reference later (Confluence) 
+- Uneven awareness and ability for interested parties to participate in decisions as they're being made (Google Docs) 
+
+We need a more inclusive, more collaborative process for making decisions that impact our codebase.
+
+## Proposed Solution
+
+RFCs will be submitted and reviewed as pull requests in this repository. If approved, they'll be stored as documentation.
+
+**Salient points in favor**
+ - Viewable by all, not just Mozilla-internals
+ - Closer to the code, easier to find
+ - In keeping with how other teams do it (TODO link evidence)
+
+
+## Guide-level explanation
+
+The high-level process to create an RFC is as follows:
+
+* Create an RFC document (like this one). (TODO: templates, maybe)
+* Open a pull request for the RFC document.
+* Ask for feedback on the pull request.
+
+During the lifetime of an RFC:
+
+* Discussions happen asynchronously on the pull request; anyone may engage.
+* Build consensus (TODO: agreement?) and integrate feedback.
```

Comment:
I think aiming for consensus is a trap here. There should be at least one key stakeholder for each RFC. We can base ourselves on the component owners and when an RFC doesn't have a clear owner fallback to the tech lead. The key stakeholder yes would need to sign off on the RFC, but overall consensus IMO should not be necessary.

Although redundant, for documentation purposes and future insurance I think it would be worth stating here that ALL comments should be addressed and resolved prior to approving an RFC, not only stakeholder comments. But when there is controversy, it should simply fallback to the key stakeholders and the RFC authors to reach an agreement.

Reply:
There is a hole in this that even applies to this RFC, lol. Who to add as a key stakeholder when the tech lead is the author and there is no clear component owner? I guess in this case no stakeholders and the tech lead just feels the vibe? I'd be fine with that.

Reply:
I think we want to aim at building consensus in the sense that all team members find the proposal acceptable enough to support it. That doesn't mean that we always have to unanimously agree that it's the best of all approaches. Just sharing, reviewing, and talking through will help a lot though in my experience, and making sure all feedback is heard and addressed.

RFC decisions can also be reversed in case the implementation doesn't pan out as expected.

My recommendation would be try it out and see how often we reach a point that requires a decision maker :). I'd expect this to be very rare.

Reply:
I agree it would be sensitive to point out that consensus is something we aim for. I also agree controversial RFCs would be the outliers. I still think the document should be clear that it's not a blocker if consensus is not reached.